### PR TITLE
[zkLogin] Add length checks and a minor change to address seed derivation

### PR DESCRIPTION
**Pull request by mskd12**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/zklogin/src/checks.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+		if (!Object.prototype.hasOwnProperty.call(this.processed, name)) {
+			throw new Error('Claim ' + name + ' not processed');
+		}
+		const details = this.processed[name];
+
+		const value_index = details.offsets.value + details.offsets.start;
+		const value_length = details.offsets.value_length;
+		if (this.decoded_payload[value_index] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
```
Comment (pchrysochoidis) [inline sdk/zklogin/src/checks.ts:176]: Could this JSON.stringify(details) contain any sensitive user data? Just worried that we might end up logging it to any logging system

File: sdk/zklogin/src/checks.ts [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+			);
+		}
+		if (this.decoded_payload[value_index + value_length - 1] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
+			);
+		}
+
+		const raw_value = this.decoded_payload.slice(value_index + 1, value_index + value_length - 1); // omit the quotes
+		if (raw_value !== details.value) {
```
Comment (pchrysochoidis) [inline sdk/zklogin/src/checks.ts:186]: Is this an error? If it is lets through if not let's remove it, no need to add this console log

File: sdk/zklogin/src/checks.ts [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    // Note: Key claim name length is being checked in genAddressSeed.
+
+	/// Are the extended claims small enough (key claim, email_verified)
+	const extendedKeyClaimLen = keyClaimDetails.ext_claim.length;
+	if (extendedKeyClaimLen > MAX_EXTENDED_KEY_CLAIM_LEN) {
+		throw new Error(`Extended key claim length is too long`);
+	}
+
+	if (keyClaimName === 'email') {
+		const evClaimDetails = processor.process('email_verified');
```
Comment (pchrysochoidis) [inline sdk/zklogin/src/checks.ts:237]: I think FB does not return `email_verified`

Comment (mskd12) [reply sdk/zklogin/src/checks.ts:237]: I see. FYI, `email_verified` is a must if the key claim name is email. That probably means we can't use email with FB.

**Review by pchrysochoidis — APPROVED** [review]

File: sdk/zklogin/tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
 	"extends": "../build-scripts/tsconfig.shared.json",
-	"include": ["src"],
+	"include": ["src", "test"],
```
Comment (pchrysochoidis) [inline sdk/zklogin/tsconfig.json:3]: do we have to include `test` here? removing it should fix the build

File: sdk/zklogin/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
 	"extends": "../build-scripts/tsconfig.shared.json",
-	"include": ["src"],
+	"include": ["src", "test/checks.test.ts"],
```
Comment (Jordan-Mysten) [inline sdk/zklogin/tsconfig.json:3]: Can you just include the test folder?

**Pull request merged**